### PR TITLE
[SPARK-31894][SS][FOLLOW-UP] Rephrase the config doc

### PR DESCRIPTION
**Pull request by xuanyuanking**
### What changes were proposed in this pull request?
Address comment in https://github.com/apache/spark/pull/28707#discussion_r461102749

### Why are the changes needed?
Hide the implementation details in the config doc.

### Does this PR introduce _any_ user-facing change?
Config doc change.

### How was this patch tested?
Document only.

### CODE REVIEW

**Comment by xuanyuanking** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1236,7 +1236,7 @@ object SQLConf {
   val STATE_STORE_FORMAT_VALIDATION_ENABLED =
     buildConf("spark.sql.streaming.stateStore.formatValidation.enabled")
       .internal()
-      .doc("When true, check if the UnsafeRow from the state store is valid or not when running " +
+      .doc("When true, check if the checkpoint from the state store is valid or not when running " +
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1239]: `checkpoint from the state store` looks a bit weird. `data from the state store`?

**Comment by cloud-fan** [comment]
do we have the same problem in the error message?

**Comment by cloud-fan** [comment]
cc @HeartSaVioR

**Comment by xuanyuanking** [comment]
https://github.com/apache/spark/pull/28707/files#diff-238c6b2f820f62c556f204788d684584R150
I think the error message is clear, we don't use code name inside.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -1236,9 +1236,9 @@ object SQLConf {
   val STATE_STORE_FORMAT_VALIDATION_ENABLED =
     buildConf("spark.sql.streaming.stateStore.formatValidation.enabled")
       .internal()
-      .doc("When true, check if the UnsafeRow from the state store is valid or not when running " +
-        "streaming queries. This can happen if the state store format has been changed. Note, " +
-        "the feature is only effective in the build-in HDFS state store provider now.")
+      .doc("When true, check if the date from state store is valid or not when running streaming " +
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1239]: data...

Comment (xuanyuanking) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: 😂

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 1 above)
Comment (xuanyuanking) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Thanks, fix done

**Comment by cloud-fan** [comment]
This just updates the config doc and we don't need to wait for the test. Thanks, merging to master!

**Comment by HeartSaVioR** [comment]
post +1. Thanks for the fix!

**Pull request closed without merging**